pkg/scm/git/git.go

```diff
@@ -29,6 +29,9 @@ type Git interface {
 	Checkout(repo, ref string) error
 	SubmoduleUpdate(repo string, init, recursive bool) error
 	GetInfo(string) *api.SourceInfo
+	CreateEmptyLocalGitDirectory() (string, error)
+	CreateLocalGitDirectory() (string, error)
+	CreateLocalGitDirectoryWithSubmodule() (string, error)
```

Comment:
What I understand:

- There is no import loop as of the commit introducing this change (41011280c32055608f28b113c5e58aa70f341720), `ToSlashCygwin` is only introduced later in 9db4e2056355bcd0ebf6f3d7c215a2a3e5c0e1f8.

- `ToSlashCygwin` is the cause of the import loop

- `ToSlashCygwin` doesn't need to belong in `pkg/scm/git/git.go`, it is something hardly related to "git".

- All calls to `ToSlashCygwin` are a no-op in non-Windows systems.

My proposal:

- Do not move stuff from `pkg/test` into `pkg/scm/git/git.go`
- Move `ToSlashCygwin` (and `isUsingCygwinGit` and `isUsingCygwinGit`) to its own package. Suggestion: `pkg/scm/git/gitpath`?

Bonus:

Use [build constraints](https://golang.org/pkg/go/build/#hdr-Build_Constraints) and get rid of the "Cygwin" part in the function name, and we could have something like:

```go
// pkg/scm/git/path/path_windows.go
// ----------------------------------

package path

func Normalize(path string) (string, error) {
       if usingCygwinGit {
               cmd := exec.Command("cygpath", path)
               out, err := cmd.Output()
               return strings.TrimRight(string(out), "\n"), err
       }
       return path, nil
}
```

```go
// pkg/scm/git/path/path.go
// ----------------------------------

// +build !windows

package path

func Normalize(path string) (string, error) {
       return path, nil
}
```

Then, wherever it gets called (from `pkg/scm/git/*`, `pkg/test/*`, etc), it would be called as:

```go
path.Normalize(...)
```

(Maybe call the package `gitpath` instead of `path`, for the latter is a stdlib package)


---

@jim-minter WDYT?